lib/aws_authenticate.py

```
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+import subprocess
+# https://docs.python.org/3.3/library/argparse.html
+import argparse
+from awscli.testutils import aws
+import time
+
+class AWSAuthenticate:
+
+  def set_aws_profile(self, mfa_token):
+    caller_identity = aws('sts get-caller-identity').json
+    mfa_devices = aws('iam list-mfa-devices').json
+    account_id = caller_identity['Account']
+    mfa_arn_of_user = mfa_devices['MFADevices'][0]['SerialNumber']
+    arn_of_role = 'arn:aws:iam::' + account_id + ':role/terraform-admin'
+
+    session = aws('sts assume-role --role-arn ' + arn_of_role + ' --role-session-name role_session --serial-number ' + mfa_arn_of_user + ' --duration-seconds 32400 --token-code ' + mfa_token)
+
+    if session.rc !=0:
+      print '\n****Something went wrong during aws assume role. Please see above log for more details.****'
```

We should throw an exception here, then catch it at the top-level and log the error message:
```python
raise Exception('****Something went wrong during aws assume role. Please see above log for more details.****')

def main():
  try:
    aa = AWSAuthenticate()
    aa.authenticate()
  except Exception as e:
    self.logger.error(e)
    return 1
```